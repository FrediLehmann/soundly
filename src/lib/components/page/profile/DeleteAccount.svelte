<script lang="ts">
  import { goto } from '$app/navigation';

  import { Button } from '$lib/components/atoms';
  import { AlertTriangle } from '$lib/Icons';
  import { supabase } from '$lib/supabase';

  let isDeleting = false;

  const deleteAccount = async () => {
    await supabase.from('profiles').update({ deleted: true });
    await supabase.auth.signOut();

    goto('/');
  };
</script>

<section>
  <h2>Delete Account</h2>
  <p>
    Once an account is delted it cannot be recovered. The deletion might take
    some time to complete. You will not necessarely be informed when the
    deletion is completed.
  </p>
  {#if isDeleting}
    <div class="warning">
      <AlertTriangle width="2rem" height="2rem" />
      <p>
        Are you sure you want to delete your account and all connected data?
        This cannot be reverted and all data will be lost forever.
      </p>
    </div>
    <Button on:click={() => (isDeleting = false)}>Cancel</Button>
    <Button style="danger" on:click={deleteAccount}>Delete</Button>
  {:else}
    <Button style="danger" on:click={() => (isDeleting = true)}
      >Delete Account</Button>
  {/if}
</section>

<style>
  h2 {
    font-size: 1.2rem;
  }

  .warning {
    display: flex;
    gap: 1rem;
    background-color: hsla(var(--secondary-100));

    border: 1px solid hsla(var(--secondary-500));
    border-radius: 0.5rem;

    padding: 0.75rem 1rem;
    margin-block-end: 1rem;
  }

  .warning > p {
    margin: 0;
  }

  section {
    max-width: 75%;
  }
</style>
