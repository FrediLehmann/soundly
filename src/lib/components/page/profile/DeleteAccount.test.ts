/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom';
import { fireEvent, render } from '@testing-library/svelte';

import DeleteAccount from './DeleteAccount.svelte';

describe('<DeleteAccount /> component', () => {
  test('renders correctly', () => {
    const { component } = render(DeleteAccount);
    expect(component).toMatchSnapshot();
  });

  test('renders edit mode correctly', async () => {
    const { component, getByRole } = render(DeleteAccount);
    await fireEvent.click(getByRole('button'));
    expect(component).toMatchSnapshot();
  });
});
