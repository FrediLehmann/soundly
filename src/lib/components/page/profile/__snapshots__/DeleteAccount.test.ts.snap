// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteAccount /> component renders correctly 1`] = `
DeleteAccount {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [
      false,
      [Function],
      [Function],
      [Function],
    ],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {},
    "root": <div>
      <section
        class="svelte-eyl0a4"
      >
        <h2
          class="svelte-eyl0a4"
        >
          Delete Account
        </h2>
         
        <p>
          Once an account is delted it cannot be recovered. The deletion might take
    some time to complete. You will not necessarely be informed when the
    deletion is completed.
        </p>
         
        <button
          class="svelte-1dsjpe6"
          data-type="danger"
          type="button"
        >
          Delete Account
        </button>
        
      </section>
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$capture_state": [Function],
  "$inject_state": [Function],
}
`;

exports[`<DeleteAccount /> component renders edit mode correctly 1`] = `
DeleteAccount {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [
      true,
      [Function],
      [Function],
      [Function],
    ],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {},
    "root": <div>
      <section
        class="svelte-eyl0a4"
      >
        <h2
          class="svelte-eyl0a4"
        >
          Delete Account
        </h2>
         
        <p>
          Once an account is delted it cannot be recovered. The deletion might take
    some time to complete. You will not necessarely be informed when the
    deletion is completed.
        </p>
         
        <div
          class="warning svelte-eyl0a4"
        >
          <div
            class="svelte-eyl0a4"
          >
            <svg
              fill="none"
              height="1.75rem"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1.75rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z"
              />
              <line
                x1="12"
                x2="12"
                y1="9"
                y2="13"
              />
              <line
                x1="12"
                x2="12.01"
                y1="17"
                y2="17"
              />
            </svg>
          </div>
           
          <p
            class="svelte-eyl0a4"
          >
            Are you sure you want to delete your account and all connected data?
        This cannot be reverted and all data will be lost forever.
          </p>
        </div>
         
        <button
          class="svelte-1dsjpe6"
          data-type="secondary"
          type="button"
        >
          Cancel
        </button>
        
         
        <button
          class="svelte-1dsjpe6"
          data-type="danger"
          type="button"
        >
          Delete
        </button>
        
      </section>
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$capture_state": [Function],
  "$inject_state": [Function],
}
`;
